### ISSUE TITLE
SC2086: ${foo:+"$foo"} incorrectly warns about needing quotes

### ISSUE THREAD

**Issue by kballard** [koalaman/shellcheck#288]
The `${parameter:+word}` construct substitutes the expansion of `word` only if `$parameter` is non-null (and substitutes nothing at all if `$var` is null or unset). If the `word` is quoted, then it will not perform word-splitting or globbing, even if the entire `${}` is not quoted.

The end result of this is the construct `${parameter:+"$parameter"}` can be used to expand to a single argument with the literal value of `$parameter` if `$parameter` is non-null, or to no arguments at all if `$parameter` is null or unset.

ShellCheck doesn't seem to realize this and claims that I need to double-quote the entire `${}`. That may be useful for other parameter expansion forms, where the existing value of `$parameter` may be substituted, but that's not the case for `:+`.

Example:

```
echo ${var:+"$var"}
     ^––SC2086 Double quote to prevent globbing and word splitting.
```

**Comment by mietek** [comment]
Related to https://github.com/koalaman/shellcheck/issues/250.

**Comment by koalaman** [comment]
Until this is fixed for real, 0a1beb8 simply ignores quote warnings on expansion where :+ is used.
